src/System.Private.ServiceModel/tests/Common/Infrastructure/OSHelper.cs

```diff
@@ -169,7 +169,39 @@ private static OSID OSIDfromOSDescription()
                 string description = pair.Item1;
                 if (osDescription.IndexOf(description, StringComparison.OrdinalIgnoreCase) >= 0)
                 {
-                    return pair.Item2;
+                    OSID detectedID =  pair.Item2;
+
+                    // A match of "AnyWindows" means we know it was Windows but don't know which version.
+                    if (detectedID == OSID.AnyWindows)
+                    {
+                        // "Microsoft Windows" is hard-coded at compilation time by RuntimeInformation.OSDescription
+                        // for win8 and netcore50 builds.  See if we had an exact match (not just prefix-match).
+                        if (string.Equals("Microsoft.Windows", CurrentOSDescription, StringComparison.OrdinalIgnoreCase))
```

Comment:
Doh!  Good catch.  No, the period is wrong.  I fixed it by pulling these hard-coded literals up into private const's to avoid this kind of type-o in the future.